protos/atomic_swap.proto

```diff
@@ -0,0 +1,69 @@
+// Copyright 2018 REMME
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ------------------------------------------------------------------------
+
+syntax = "proto3";
+
+message AtomicSwapMethod {
+    enum Method {
+        INIT = 0;
+        APPROVE = 1;
+        EXPIRE = 2;
+        SET_SECRET_LOCK = 3;
+        CLOSE = 4;
+    }
+}
+
+message AtomicSwapInitPayload {
+    string receiver_address = 1;
+    string sender_address_non_local = 7;
+    uint64 amount = 2;
+    string swap_id = 3;
+    string secret_lock_optional_bob = 4;
+    string email_address_encrypted_optional_alice = 5;
+    uint32 created_at = 6;
+}
+
+message AtomicSwapApprovePayload {
+    string swap_id = 1;
+}
+
+message AtomicSwapExpirePayload {
+    string swap_id = 1;
+}
+
+message AtomicSwapSetSecretLockPayload {
+    string swap_id = 1;
+    string secret_lock = 2;
+}
+
+message AtomicSwapClosePayload {
+    string swap_id = 1;
+    string secret_key = 2;
+}
+
+message AtomicSwapInfo {
```

Comment:
There is a lot of duplicated fields. Maybe a composition of values would be appropriate?

Reply:
Would you like to give an example and how it will ease the development process and readability?

Reply:
At least we will not have duplicated fields. I believe they are some metadata related to the whole atomic swap process. Why not mode them out to a separate object nested into the corresponding top-level objects. @apadenko what do you think about it?

Reply:
I believe, composition will give more readability. But It's not required at this case. 
At least, for example, we can do something like. 

```
message AtomicSwapBasePayload {
    uint64 amount = 1;
    string swap_id = 2;
    uint32 created_at = 3;
    string receiver_address = 4;
    string sender_address_non_local = 5;
}

message AtomicSwapInitPayload {
    AtomicSwapBasePayload baseInfo = 1;
    string secret_lock_optional_bob = 4;
    string email_address_encrypted_optional_alice = 5;
    uint32 created_at = 6;
}

message AtomicSwapInfo {
    AtomicSwapBasePayload baseInfo = 1;
    bool is_closed = 1;
    bool is_approved = 11;
    string sender_address = 2;
    string email_address_encrypted_optional = 5;
    string secret_lock = 7;
    string secret_key = 8;
    bool is_initiator = 10; // isAlice
}
```

Or somethig kind of that. 
Also I have a question, what is the difference between 
```sender_address_non_local ```
 and 
```sender_address ``` ?

Reply:
I think we need documentation for every single field :)